File: backend/app/schemas/anexo.py
"""Schemas de Anexo: payloads de entrada e saída de `app/api/routes/anexo.py`.

Sem `AnexoUpdate` - decisão confirmada explicitamente com o usuário antes da
implementação: Anexo é create + read + soft-delete apenas, sem PATCH (mesmo
raciocínio de Financiamento/Empréstimo - ver
docs/analise-arquitetural-anexo.md).
"""
from datetime import datetime

from pydantic import BaseModel, Field, field_validator

from app.schemas.base import OrmBaseModel


class AnexoCreate(BaseModel):
    transacao_id: int
    nome_original: str = Field(min_length=1, max_length=255)
    caminho_arquivo: str = Field(min_length=1, max_length=500)
    mime_type: str | None = Field(default=None, max_length=100)
    tamanho_bytes: int | None = Field(default=None, ge=0)

    @field_validator("nome_original", "caminho_arquivo", mode="before")
    @classmethod
    def _remover_espacos_nas_pontas(cls, valor: str) -> str:
        # mode="before": mesmo raciocínio já usado em Tag/Cartão/Meta - roda
        # ANTES do min_length ser conferido, senão um valor só de espaços
        # passaria a checagem de tamanho e só viraria vazio depois.
        return valor.strip() if isinstance(valor, str) else valor


class AnexoRead(OrmBaseModel):
    id: int
    transacao_id: int
    nome_original: str
    caminho_arquivo: str
    mime_type: str | None
    tamanho_bytes: int | None
    data_upload: datetime
    ativo: bool

File: frontend/src/types/anexo.ts
/**
 * Espelha 1:1 `app/schemas/anexo.py` — conferido por leitura direta do
 * backend. Sem `AnexoUpdate`: Anexo é create + read + soft-delete apenas,
 * decisão já confirmada no backend (ver docs/analise-arquitetural-anexo.md).
 *
 * `caminho_arquivo` é só uma referência textual (caminho local, URL, chave
 * de storage externo) — o backend nunca armazena o binário em si (ver
 * docs/analise-arquitetural-anexo-frontend.md).
 */
export interface AnexoRead {
  id: number;
  transacao_id: number;
  nome_original: string;
  caminho_arquivo: string;
  mime_type: string | null;
  tamanho_bytes: number | null;
  data_upload: string;
  ativo: boolean;
}

export interface AnexoCreate {
  transacao_id: number;
  nome_original: string;
  caminho_arquivo: string;
  mime_type?: string | null;
  tamanho_bytes?: number | null;
}
